From f92c37c9fc6f17f264198b32cad7abf3969bd6e4 Mon Sep 17 00:00:00 2001
From: Li Jun <B47624@freescale.com>
Date: Mon, 29 Sep 2014 10:32:00 +0800
Subject: [PATCH 0131/5242] MLK-9618-3 usb: chipidea: otg: enable hnp_pollinig
 flag

commit  ff9f0c893f639e409e31fd0f6b2595f9e65314b1 from
https://source.codeaurora.org/external/imx/linux-imx.git

This patch enables hnp_polling flag for chipidea host driver in otg fsm mode.

Signed-off-by: Li Jun <b47624@freescale.com>
(cherry picked from commit acb43a932263edd9b80c6d6d9ee858176c201542)
(cherry picked from commit 9e75e851370b62319fec4200b68cfd3465e12cfe)
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/usb/chipidea/otg_fsm.c |    1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/usb/chipidea/otg_fsm.c b/drivers/usb/chipidea/otg_fsm.c
index fa85e12..8180c9c 100644
--- a/drivers/usb/chipidea/otg_fsm.c
+++ b/drivers/usb/chipidea/otg_fsm.c
@@ -882,6 +882,7 @@ int ci_hdrc_otg_fsm_init(struct ci_hdrc *ci)
 	ci->otg.gadget = &ci->gadget;
 	ci->fsm.otg = &ci->otg;
 	ci->fsm.power_up = 1;
+	ci->fsm.hnp_polling = 1;
 	ci->fsm.id = hw_read_otgsc(ci, OTGSC_ID) ? 1 : 0;
 	ci->fsm.otg->state = OTG_STATE_UNDEFINED;
 	ci->fsm.ops = &ci_otg_ops;
-- 
1.7.9.5

